### PR TITLE
Introduce the four-color pen

### PR DESCRIPTION
**Pull request by octareenroon91**
Add a fancy pen that lets you choose between black, red, green, and blue ink.

:cl:
rscadd: Add the four-color pen, which writes in black, red, green, and blue.
rscadd: Adds two such pens to the Bureaucracy supply pack.
/:cl:

### CODE REVIEW

File: code/modules/paperwork/pen.dm [hunk 1]
```diff
@@ -44,6 +44,31 @@
...
+		if("black")
+			colour = "red"
+		if("red")
+			colour = "green"
+		if("green")
+			colour = "blue"
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [src.colour].</span>"
+	//user << "<span class='notice'>[src] will now write in <span style='color=[colour]'>[colour]</span>.</span>" // If anyone feels lucky
```
Comment (Anonus) [inline code/modules/paperwork/pen.dm]: not span style, font style!

Comment (optimumtact) [inline code/modules/paperwork/pen.dm]: also remove this dead comment

**Comment by MrStonedOne** [comment]
New content, Please make a changelog.

You can include a changelog in the body of the pr now: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=5157

File: code/modules/paperwork/pen.dm (hunk 1 above)
Comment (octareenroon91) [inline code/modules/paperwork/pen.dm]: That will make the colored feedback work?

**Comment by Iamgoofball** [comment]
why not make it allow a color hex?

**Comment by octareenroon91** [comment]
I wasn't sure how hard it was to pull up the color hex interface.
Also the four-color pen is iconic, and should exist whether or not a more expansive pen does.

**Comment by YotaXP** [comment]
You can bring up the color picker with `input(...)as color`, but I agree that 4-color is better.

File: code/modules/paperwork/pen.dm (hunk 1 above)
Comment (Anonus) [inline code/modules/paperwork/pen.dm]: Yes it will.

**Comment by WJohn** [comment]
Where's the "awww" tag?

**Comment by octareenroon91** [comment]
What purpose would that serve?

**Comment by WJohn** [comment]
There's so many adorable additions that should have "awww" tags.

**Comment by Cheridan** [comment]
Too OP. This instantly invalidates all other pens! Think of the game's delicate balance.

File: code/modules/paperwork/pen.dm [hunk 2]
```diff
@@ -44,6 +44,31 @@
...
+/obj/item/weapon/pen/fourcolor
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "pen"
+	item_state = "pen"
+
+/obj/item/weapon/pen/fourcolor/New()
+	..()
+	colour = "black"
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: just set these by default

File: code/modules/paperwork/pen.dm [hunk 3]
```diff
@@ -44,6 +44,31 @@
...
+	desc = "It's a fancy four-color ink pen, set to [src.colour]."
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: set this by default, swapping ```[src.colour]``` here with "black" to match the line above.

File: code/modules/paperwork/pen.dm [hunk 4]
```diff
@@ -44,6 +44,31 @@
...
+	switch(colour)
+		if("black")
+			colour = "red"
+		if("red")
+			colour = "green"
+		if("green")
+			colour = "blue"
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [src.colour].</span>"
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: remove ```src.```

File: code/modules/paperwork/pen.dm [hunk 5]
```diff
@@ -44,6 +44,31 @@
...
+	//user << "<span class='notice'>[src] will now write in <span style='color=[colour]'>[colour]</span>.</span>" // If anyone feels lucky
+	desc = "It's a fancy four-color ink pen, set to [src.colour]."
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: remove ```src.```

File: code/modules/paperwork/pen.dm [hunk 6]
```diff
@@ -44,6 +44,31 @@
 	icon_state = "pen"
 	colour = "white"
 
+/obj/item/weapon/pen/fourcolor
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: this is inherited, remove this line

File: code/modules/paperwork/pen.dm [hunk 7]
```diff
@@ -44,6 +44,31 @@
 	icon_state = "pen"
 	colour = "white"
 
+/obj/item/weapon/pen/fourcolor
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "pen"
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: this is inherited, remove this line

File: code/modules/paperwork/pen.dm [hunk 8]
```diff
@@ -44,6 +44,31 @@
 	icon_state = "pen"
 	colour = "white"
 
+/obj/item/weapon/pen/fourcolor
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "pen"
+	item_state = "pen"
```
Comment (RemieRichards) [inline code/modules/paperwork/pen.dm]: this is inherited, remove this line

**Comment by octareenroon91** [comment]
How about if there's only one and it spawns on the captain?

**Comment by GunHog** [comment]
Pretty sure he was joking. Maybe.

File: code/modules/paperwork/pen.dm (hunk 7 above)
Comment (octareenroon91) [inline code/modules/paperwork/pen.dm]: the invisible pen retains its duplicate icon_state, though

File: code/modules/paperwork/pen.dm [hunk 9]
```diff
@@ -44,6 +44,24 @@
...
+		if("black")
+			colour = "red"
+		if("red")
+			colour = "green"
+		if("green")
+			colour = "blue"
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [colour].</span>"
+	//user << "<span class='notice'>[src] will now write in <font style='color=[colour]'>[colour]</font>.</span>"
```
Comment (optimumtact) [inline code/modules/paperwork/pen.dm]: REMOVE IT

**Comment by Cheridan** [comment]
@GunHog He actually tagged it, the absolute madman

**Comment by duncathan** [comment]
I will not rest until bureaucracy is 100% CSGO balanced

**Comment by YotaXP** [comment]
Librarian could use a buff.  How about letting their PDA start with one of these bad boys?

**Comment by RemieRichards** [comment]
@YotaXP I dont think you realise how absurd you sound right now... the Librarian has been a main staple of  teams for a long time now.

**Pull request merged**